types/index.d.ts

```diff
@@ -13,20 +13,24 @@ export interface ChalkOptions {
 	level?: Level;
 }
 
-export interface Chalk {
+export interface ChalkConstructor {
 	new (options?: ChalkOptions): Chalk;
 	(options?: ChalkOptions): Chalk;
+}
+
+export interface ColorSupport {
+	level: Level;
+	hasBasic: boolean;
+	has256: boolean;
+	has16m: boolean;
+}
+
+export interface Chalk {
 	(...text: string[]): string;
 	(text: TemplateStringsArray, ...placeholders: string[]): string;
-	constructor: Chalk;
+	constructor: ChalkConstructor;
 	enabled: boolean;
 	level: Level;
-	supportsColor: {
-		level: Level;
-		hasBasic: boolean;
-		has256: boolean;
-		has16m: boolean;
-	};
 	rgb(r: number, g: number, b: number): Chalk;
 	hsl(h: number, s: number, l: number): Chalk;
```

Comment:
Instead of defining that all these methods return `Chalk`, you can also say that they return `this`.

```js
rgb(r: number, g: number, b: number): this;
```

That's also what TS generates for the type definitions. Just another approach though.

Reply:
Updated